src/Form/index.tsx

```diff
@@ -1868,8 +1869,30 @@ function Form({
           break;
         }
       } else if (type === ACTION_OPEN_FUSER_ENVELOPES) {
-        await client.generateEnvelopes(action);
-        openTab(getSignUrl());
+        try {
+          // waiting 4 seconds for documents to generate before redirect
+          setTimeout(() => {
+            openTab(getSignUrl());
+          }, 4000);
+        } catch (e: any) {
+          setElementError((e as Error).message);
+          break;
+        }
+      } else if (type === ACTION_GENERATE_QUIK_ENVELOPES) {
+        try {
+          const newValues = await client.generateQuikEnvelopes(action);
+          // waiting 4 seconds for documents to generate before redirect
+          if (action.form_fill_type === 'pdf') {
+            setTimeout(() => {
```

Comment:
this if statement isn't needed. if the user wants to sign documents, they can add a "Sign Documents" action separately. you can just run

updateFieldValues(newValues)